src/lsm/lsm_merge.c

```diff
@@ -283,9 +279,9 @@ __lsm_merge_span(WT_SESSION_IMPL *session, WT_LSM_TREE *lsm_tree,
 			WT_ASSERT(session,
 			    F_ISSET(youngest, WT_LSM_CHUNK_MERGING));
 			F_CLR(youngest, WT_LSM_CHUNK_MERGING);
-			record_count -= youngest->count;
-			chunk_size -= youngest->size;
+			/* try again with smaller range */
 			--end_chunk;
+			goto retry_find;
```

Comment:
Using this goto path here makes me nervous - in the other case where we restart the search any altered state has been reset.

Reply:
i don't like such update either. However, ```youngest_gen``` and ```oldest_gen``` would represent obsolete value without go back to the original ```retry_find ``` place. it's not that easy to restore them like what you did on ```record_count ``` and ```chunk_size```.

Reply:
I'd probably pull the six lines here:
https://github.com/wiredtiger/wiredtiger/blob/develop/src/lsm/lsm_merge.c#L311
out into a function called `__lsm_merge_clear(WT_SESSION_IMPL *session, WT_LSM_TREE *lsm_tree, u_int start, u_int nchunks)` and call that before the `goto retry_find` as well as in the location below.

My fear is that a bug in this code will leave a chunk with the `WT_LSM_CHUNK_MERGING` flag set, which would lead it to never be merged.